### PR TITLE
Pgvector fixes

### PR DESCRIPTION
**Pull request by edlouth**
Fixes for https://github.com/vanna-ai/vanna/issues/659

@andreped could you take a look?

### CODE REVIEW

File: src/vanna/pgvector/pgvector.py
Comment (andreped) [inline src/vanna/pgvector/pgvector.py:255]: If you remove these naively, you will run into this:
```
E       TypeError: Can't instantiate abstract class PG_VectorStore with abstract methods assistant_message, submit_prompt, system_message, user_message
```

So more code changes are required if you do not want these.

**Review by andreped — CHANGES_REQUESTED** [review]

File: src/vanna/pgvector/pgvector.py
Comment (edlouth) [reply src/vanna/pgvector/pgvector.py:255]: I was copying https://github.com/vanna-ai/vanna/blob/main/src/vanna/vannadb/vannadb_vector.py

I need this change, otherwise I get a different error as the response is `None`.  But I know what you mean, this seems to be a feature of inheriting from `VannaBase` for both vector stores and LLM models.

Comment (edlouth) [reply src/vanna/pgvector/pgvector.py:255]: Looking into it I think the solution is to import the two classes the other way around

```
class CustomVanna(OpenAI_Chat, PG_VectorStore):
    def __init__(self, config=None):
        OpenAI_Chat.__init__(self, config=config)
        PG_VectorStore.__init__(self, config=config)
```

So I can potentially revert this change, although I was basing the import order on the [docs](https://vanna.ai/docs/postgres-openai-standard-qdrant/)

Comment (andreped) [reply src/vanna/pgvector/pgvector.py:255]: I found the issue with the default embedding model. I will make a PR to your branch.

Comment (andreped) [reply src/vanna/pgvector/pgvector.py:255]: I made a PR to your branch now:
https://github.com/edlouth/vanna/pull/1

I have also made changes to the unit test, which I will make a separate PR to your branch for.

Comment (edlouth) [reply src/vanna/pgvector/pgvector.py]: Thanks for sorting that. I have merged the PR on my branch, so it is in this PR now.

Comment (andreped) [reply src/vanna/pgvector/pgvector.py]: I made another PR now, which reverts the methods you removed again, as they are not needed.
I also updated the PGVector unit tests:
https://github.com/edlouth/vanna/pull/2

Comment (andreped) [reply src/vanna/pgvector/pgvector.py]: I made another PR now, which reverts the methods you removed again, as they are not needed.
I also updated the PGVector unit tests:
https://github.com/edlouth/vanna/pull/2

**Comment by edlouth** [comment]
All of @andreped's changes are now merged into this PR and I think it is ready to merge.

**Comment by andreped** [comment]
> All of @andreped's changes are now merged into this PR and I think it is ready to merge.

@edlouth Did you manage to run the pgvector tests? If not, could you do so, to make sure this PR is good?
```
pytest -v tests/test_pgvector.py
```

**Comment by edlouth** [comment]
> > All of @andreped's changes are now merged into this PR and I think it is ready to merge.
> 
> @edlouth Did you manage to run the pgvector tests? If not, could you do so, to make sure this PR is good?
> 
> ```
> pytest -v tests/test_pgvector.py
> ```

The tests are passing for me, I made a small change to bring in the database port from an environment variable.

**Comment by andreped** [comment]
> The tests are passing for me, I made a small change to bring in the database port from an environment variable.

Great! Then I think PR is good!

Still requires approval from @zainhoda before merge though, but great job, @edlouth! :]

**Review by andreped — APPROVED** [review]
LGTM!

**Review by zainhoda — APPROVED** [review]

**Comment by zainhoda** [comment]
I commented out the tests because I'm getting a bunch of dependency issues trying to run them. We can add them back later

**Pull request merged**
